rust/relay/src/main.rs

```diff
@@ -200,27 +232,39 @@ impl<R> Eventloop<R>
 where
     R: Rng,
 {
-    async fn new(
+    fn new(
         server: Server<R>,
         channel: Option<PhoenixChannel<InboundPortalMessage, ()>>,
-        listen_ip4_address: Ipv4Addr,
+        listen_address: IpStack,
         _: &mut Registry,
     ) -> Result<Self> {
         let (relay_data_sender, relay_data_receiver) = mpsc::channel(1);
         let (inbound_data_sender, inbound_data_receiver) = mpsc::channel(10);
-        let (outbound_data_sender, outbound_data_receiver) =
+        let (outbound_ip4_data_sender, outbound_ip4_data_receiver) =
```

Comment:
There isn't but it doesn't really matter either, right? This way, there are less conditionals in the code which I find easier to understand :)

Reply:
I'd have to make them an `Option` in the event loop then which creates an additional error path when trying to send to them in case we need to.

This way, the path we will never try to use the channel, assuming `Server` doesn't have a bug in how it selects the relay addresses :)